server/src/main/java/org/apache/druid/curator/announcement/NodeAnnouncer.java

```
@@ -0,0 +1,346 @@
+      }
+
+      if (!parentsIBuilt.isEmpty()) {
+        CuratorTransaction transaction = curator.inTransaction();
+        for (String parent : parentsIBuilt) {
+          try {
+            transaction = transaction.delete().forPath(parent).and();
+          }
+          catch (Exception e) {
+            log.info(e, "Unable to delete parent[%s], boooo.", parent);
+          }
+        }
+        try {
+          ((CuratorTransactionFinal) transaction).commit();
+        }
+        catch (Exception e) {
+          log.info(e, "Unable to commit transaction. Please feed the hamsters");
+        }
+      }
+    }
+  }
+
+  /**
+   * Like announce(path, bytes, true).
+   */
+  public void announce(String path, byte[] bytes)
+  {
+    announce(path, bytes, true);
+  }
+
+  /**
+   * Announces the provided bytes at the given path.  Announcement means that it will create an ephemeral node
+   * and monitor it to make sure that it always exists until it is unannounced or this object is closed.
+   *
+   * @param path                  The path to announce at
+   * @param bytes                 The payload to announce
+   * @param removeParentIfCreated remove parent of "path" if we had created that parent
+   */
+  public void announce(String path, byte[] bytes, boolean removeParentIfCreated)
+  {
+    synchronized (toAnnounce) {
+      if (!started) {
+        toAnnounce.add(new Announceable(path, bytes, removeParentIfCreated));
+        return;
+      }
+    }
+
+    final ZKPaths.PathAndNode pathAndNode = ZKPaths.getPathAndNode(path);
+
+    final String parentPath = pathAndNode.getPath();
+    boolean buildParentPath = false;
+
+    byte[] value = announcements.get(path);
+
+    if (value == null) {
+      try {
+        if (curator.checkExists().forPath(parentPath) == null) {
+          buildParentPath = true;
+        }
+      }
+      catch (Exception e) {
+        log.debug(e, "Problem checking if the parent existed, ignoring.");
```

Why ignoring?

---

this part of code is following the original Announcer's code.

when this exception happens: 1, if the parent path exists, this exception can be ignored. 2 if the parent path doesn't exist, the following code will throw exception due to failed to add path